File: SARAssist/SARAnnotation.m
//
//  SARAnnotation.m
//  SARAssist
//
//  Created by V-FEXrt on 7/18/15.
//  Copyright (c) 2015 V-FEXrt. All rights reserved.
//

#import "SARAnnotation.h"

@implementation SARAnnotation

- (instancetype)initWithTitle:(NSString *)title Location:(CLLocationCoordinate2D)location Color:(MKPinAnnotationColor)color
{
    self = [super init];
    if (self) {
        self.title = title;
        self.coordinate = location;
        self.color = color;
    }
    return self;
}
-(MKPinAnnotationView *)annotationView{
    MKPinAnnotationView *annotationView = [[MKPinAnnotationView alloc]initWithAnnotation:self reuseIdentifier:@"CustomPinAnnotationView"];
    
    UIButton * btn = [UIButton buttonWithType:UIButtonTypeInfoDark];
    [btn addTarget:self action:@selector(callDelegate) forControlEvents:UIControlEventTouchUpInside];
    
    annotationView.enabled = YES;
    annotationView.canShowCallout = YES;
    annotationView.pinColor = self.color;
    annotationView.rightCalloutAccessoryView = btn;
    
    return annotationView;
    
}

-(void)callDelegate{
    [self.delegate buttonWasPressedOnSARAnnotation:self];
}
@end

File: SARAssist_Swift/SARAssist/SARAnnotation.swift
//
//  SARAnnotation.swift
//  SARAssist
//
//  Created by V-FEXrt on 9/19/15.
//  Copyright © 2015 V-FEXrt. All rights reserved.
//

import UIKit
import MapKit

protocol SARAnnotationDelegate{
    func annotationWasSelected(annotation:SARAnnotation)
}

class SARAnnotation: NSObject, MKAnnotation {
    
    @objc var coordinate: CLLocationCoordinate2D
    @objc var title: String?
    internal var color:MKPinAnnotationColor
    internal var delegate:SARAnnotationDelegate?
    internal var row:Int = 0
    internal var column:Int = 0
    
    init(coordinate: CLLocationCoordinate2D, title: String, color: MKPinAnnotationColor) {
        self.coordinate = coordinate
        self.title = title
        self.color = color
    }
    
    func annotationView() -> MKPinAnnotationView{
        let annotationView:MKPinAnnotationView = MKPinAnnotationView(annotation: self, reuseIdentifier: "CustomPinAnnotationView")
        let btn:UIButton = UIButton(type: .InfoDark)
        btn.addTarget(self, action: Selector("callDelegate"), forControlEvents: .TouchUpInside)
        
        annotationView.enabled = true
        annotationView.canShowCallout = true
        annotationView.pinColor = self.color
        annotationView.rightCalloutAccessoryView = btn
        
        return annotationView
    }
    
    func callDelegate(){
        self.delegate?.annotationWasSelected(self);
    }

}
